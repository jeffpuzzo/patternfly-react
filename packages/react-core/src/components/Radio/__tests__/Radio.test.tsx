import React from 'react';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import { Radio } from '../Radio';

const props = {
  onChange: jest.fn()
};

describe('Radio', () => {
  test('controlled', () => {
    render(<Radio isChecked id="check" aria-label="check" name="check" />);
    expect(screen.getByRole('radio').outerHTML).toMatchSnapshot();
  });

  test('uncontrolled', () => {
    render(<Radio id="check" aria-label="check" name="check" />);
    expect(screen.getByRole('radio').outerHTML).toMatchSnapshot();
  });

  test('isDisabled', () => {
    render(<Radio id="check" isDisabled aria-label="check" name="check" />);
    expect(screen.getByRole('radio').outerHTML).toMatchSnapshot();
  });

  test('isLabelBeforeButton', () => {
    render(<Radio id="check" isLabelBeforeButton label="Radio label" aria-label="check" name="check" />);
    expect(screen.getByRole('radio').outerHTML).toMatchSnapshot();
  });

  test('isLabelWrapped', () => {
    render(<Radio id="check" isLabelWrapped label="Radio label" aria-label="check" name="check" />);
    expect(screen.getByRole('radio').outerHTML).toMatchSnapshot();
  });

  test('label is string', () => {
    render(<Radio label="Label" id="check" isChecked aria-label="check" name="check" />);
    expect(screen.getByRole('radio').outerHTML).toMatchSnapshot();
  });

  test('label is function', () => {
    const functionLabel = () => <h1>Header</h1>;
    render(<Radio label={functionLabel()} id="check" isChecked aria-label="check" name="check" />);
    expect(screen.getByRole('radio').outerHTML).toMatchSnapshot();
  });

  test('label is node', () => {
    render(<Radio label={<h1>Header</h1>} id="check" isChecked aria-label="check" name="check" />);
    expect(screen.getByRole('radio').outerHTML).toMatchSnapshot();
  });

  test('passing class', () => {
    render(<Radio label="label" className="class-123" id="check" isChecked aria-label="check" name="check" />);
    expect(screen.getByRole('radio').outerHTML).toMatchSnapshot();
  });

  test('passing HTML attribute', () => {
    render(<Radio label="label" aria-labelledby="labelId" id="check" isChecked aria-label="check" name="check" />);
    expect(screen.getByRole('radio').outerHTML).toMatchSnapshot();
  });

  test('Radio passes value and event to onChange handler', () => {
    render(<Radio id="check" {...props} aria-label="check" name="check" />);

    userEvent.click(screen.getByRole('radio'));
    expect(props.onChange).toHaveBeenCalled();
  });

  test('Radio description', () => {
    render(<Radio id="check" name="check" aria-label="check" description="Text description..." />);
    expect(screen.getByText('Text description...')).toBeInTheDocument();
  });

  test('Radio body', () => {
    render(<Radio id="check" name="check" aria-label="check" body="Text body..." />);
    expect(screen.getByText('Text body...')).toBeInTheDocument();
  });

  test('should throw console error when no id is given', () => {
    const myMock = jest.fn();
    global.console = { ...global.console, error: myMock };

    render(<Radio id={undefined} name="check" aria-label="check" description="Text description..." />);

    expect(myMock).toBeCalled();
  });
});
