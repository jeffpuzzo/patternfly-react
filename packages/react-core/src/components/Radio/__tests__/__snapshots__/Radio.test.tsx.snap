// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio controlled 1`] = `"<input id=\\"check\\" name=\\"check\\" class=\\"pf-c-radio__input\\" type=\\"radio\\" aria-invalid=\\"false\\" aria-label=\\"check\\" data-ouia-component-type=\\"PF4/Radio\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Radio-1\\" checked=\\"\\">"`;

exports[`Radio isDisabled 1`] = `"<input id=\\"check\\" name=\\"check\\" class=\\"pf-c-radio__input\\" type=\\"radio\\" aria-invalid=\\"false\\" disabled=\\"\\" aria-label=\\"check\\" data-ouia-component-type=\\"PF4/Radio\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Radio-3\\">"`;

exports[`Radio isLabelBeforeButton 1`] = `"<input id=\\"check\\" name=\\"check\\" class=\\"pf-c-radio__input\\" type=\\"radio\\" aria-invalid=\\"false\\" data-ouia-component-type=\\"PF4/Radio\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Radio-4\\">"`;

exports[`Radio isLabelWrapped 1`] = `"<input id=\\"check\\" name=\\"check\\" class=\\"pf-c-radio__input\\" type=\\"radio\\" aria-invalid=\\"false\\" data-ouia-component-type=\\"PF4/Radio\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Radio-5\\">"`;

exports[`Radio label is function 1`] = `"<input id=\\"check\\" name=\\"check\\" class=\\"pf-c-radio__input\\" type=\\"radio\\" aria-invalid=\\"false\\" data-ouia-component-type=\\"PF4/Radio\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Radio-7\\" checked=\\"\\">"`;

exports[`Radio label is node 1`] = `"<input id=\\"check\\" name=\\"check\\" class=\\"pf-c-radio__input\\" type=\\"radio\\" aria-invalid=\\"false\\" data-ouia-component-type=\\"PF4/Radio\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Radio-8\\" checked=\\"\\">"`;

exports[`Radio label is string 1`] = `"<input id=\\"check\\" name=\\"check\\" class=\\"pf-c-radio__input\\" type=\\"radio\\" aria-invalid=\\"false\\" data-ouia-component-type=\\"PF4/Radio\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Radio-6\\" checked=\\"\\">"`;

exports[`Radio passing HTML attribute 1`] = `"<input aria-labelledby=\\"labelId\\" id=\\"check\\" name=\\"check\\" class=\\"pf-c-radio__input\\" type=\\"radio\\" aria-invalid=\\"false\\" data-ouia-component-type=\\"PF4/Radio\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Radio-10\\" checked=\\"\\">"`;

exports[`Radio passing class 1`] = `"<input id=\\"check\\" name=\\"check\\" class=\\"pf-c-radio__input\\" type=\\"radio\\" aria-invalid=\\"false\\" data-ouia-component-type=\\"PF4/Radio\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Radio-9\\" checked=\\"\\">"`;

exports[`Radio uncontrolled 1`] = `"<input id=\\"check\\" name=\\"check\\" class=\\"pf-c-radio__input\\" type=\\"radio\\" aria-invalid=\\"false\\" aria-label=\\"check\\" data-ouia-component-type=\\"PF4/Radio\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Radio-2\\">"`;
